### DISCUSSION TITLE
Overclocking of intel CPU

### DISCUSSION THREAD

**Discussion by nikitavedre** [cyring/CoreFreq#427] [Q&A]
I have Intel(R) Core(TM) i7-8650U CPU.I want to set it's frequency. How to do that? Do you know MSR register to set frequency?

**Comment by cyring** [comment]
Please post here your Processor data using 
```
corefreq-cli -s -n -m -n -B -n -M
```

**Comment by nikitavedre** [comment]
can you give example of abbreviation corefreq-cli -s -n -m -n -B -n -M ?

**Reply by cyring** [reply]
> can you give example of abbreviation corefreq-cli -s -n -m -n -B -n -M ?

Is this the first time you are using _CoreFreq_ ?

**Reply by nikitavedre** [reply]
yes

**Reply by cyring** [reply]
> Do you know MSR register to set frequency?

Not frequency in Hertz directly but P-State and frequency **ratio**

There are various registers to set; you would start with the Target P-State and tune the Turbo ratios

https://github.com/cyring/CoreFreq/blob/1b52ce5f2ce88d3619c74dd0647fd16de35fce9b/intel_reg.h#L704

https://github.com/cyring/CoreFreq/blob/1b52ce5f2ce88d3619c74dd0647fd16de35fce9b/intel_reg.h#L71

**Reply by nikitavedre** [reply]
I have used MSR_NHM_TURBO_RATIO_LIMIT. But it is not setting frequency. Is there another register to set frequency?

**Reply by cyring** [reply]
What about `IA32_PERF_CTL(0x199)` ?

Remark: Per Core MSR registers have to be written with same settings.

**Reply by nikitavedre** [reply]
IA32_PERF_CTL(0x199) is also not working.Value get reset after setting

**Reply by cyring** [reply]
> IA32_PERF_CTL(0x199) is also not working.Value get reset after setting

Because another kernel driver is altering values afterward.
Prevent drivers from loading and you will have the full control over the P-States
See example in [Wiki](https://github.com/cyring/CoreFreq/wiki/CoreFreq-as-the-Clock-Source,-CPU-Freq-and-CPU-Idle-driver) to blacklist drivers.

Then CPU-Freq is confirmed not managed:
![2023-03-14-135544_644x564_scrot](https://user-images.githubusercontent.com/11563789/225007405-7bdcb946-6807-41e9-8c63-e38ca6fb0681.png)

You can now do your MSR settings which should be persistent.

**Reply by cyring** [reply]
And _CoreFreq_ is made for this:

1. Register its CPU-Freq, CPU-Idle, Governor from the `Settings` window

![2023-03-14-140526_644x1012_scrot](https://user-images.githubusercontent.com/11563789/225009724-bb886f96-3888-45a6-becf-f34f9c59383f.png)

2. Alter the Ratio from the `Processor` window, section `Performance TGT`

![2023-03-14-140708_644x1012_scrot](https://user-images.githubusercontent.com/11563789/225010236-227a39a7-f82e-4d38-b9a1-a2aad4946b97.png)

**Comment by nikitavedre** [comment]
can this repository work on windows?

**Reply by cyring** [reply]
Not at all.

**Reply by nikitavedre** [reply]
Do you know any way which can use for windows platform?

**Reply by cyring** [reply]
Depending of your needs, Windows has already numerous programs. 
# Open source 
* Intel [pcm](https://github.com/intel/pcm) 
* [ZenTimings](https://github.com/irusanov/ZenTimings)
# Binaries 
* hwmon
* cpuz
* occt
* hwinfo
* Intel XTU 
* AMD Ryzen Master 
* all other vendor's software
# BIOS, UEFI
## Open source 
* Hardware Detection Tool [HDT](https://git.zytor.com/syslinux/syslinux.git/tree/com32/hdt) 
* [Memtest86+](https://github.com/memtest86plus/memtest86plus)
